### PR TITLE
Refactors human explosion damage, rebalance explosions and bomb armor

### PR DESCRIPTION
**Pull request by datlo**
## What Does This PR Do

This PR refactors and rebalances human mobs ex_act, mainly aimed at making explosions less effective on armored targets and less random in their effectiveness.

- Explosions no longer put you to sleep.
- Explosions reliably stun, with a duration variable on bomb armor and explosion severity.
- Bomb armor now protects limbs from being severed. At max armor (bomb suit, admin gear), delimbing is impossible. At lower armor, delimbing chance is reduced but still possible. This is based on the limb's armor, so a chest piece will not help prevent delimbing.
- Number of limbs removed is now based on explosion severity, instead of being random. Which limb is affected also varies, see below for details. Overall this should result in less limbs being removed at low severity or on high armor target.
- Bomb armor now reliably lowers the damage you take from explosions, with only half damage taken with a full suit.

<details>
<summary>
<i>Details of explosion changes for each severity</i>
</summary>
<p>

Devastating severity (On the same tile of a syndi minibomb, or close to a large bomb) : 
- Chance of gibbing is still based on bomb armor. Bomb suit = Gibbing impossible, unchanged from live.
- Damage if not gibbed is now 200 brute, 200 burn, from 500 brute
- Limbs removed : 50 percent chance to sever for each arm and each leg, reduced by bomb armor.

Heavy Severity (Right next to a minibomb, or a bit further from a large bomb)
- Damage is now reliably reduced by bomb armor. Maximum damage : 60 brute, 60 burn, minimum damage : 30 brute, 30 burn (Current damage is 60 brute 60 burn, with a chance to half it based on armor)
- Now stuns for 20 seconds instead of knocking out for 20 seconds. Duration is reduced based on bomb resistance, with a minimum of 4 seconds.
- Limbs removed : 50 percent chance to lose either an arm or a leg, reduced by bomb armor.

Light Severity (Close to a minibomb, or even further away from a large bomb)
- Maximum damage : 30 brute, minimum damage : 15 brute. (Current damage is 30 brute, with a chance to half it based on armor)
- Now stuns for 16 seconds instead of knocking out for 20 seconds. Duration is reduced based on bomb resistance, bomb suit wearers will not be stunned.
- Limbs removed : 25 percent chance to lose either a hand or a foot, reduced by bomb armor.

</p>
</details>

Significant balance changes that result from these : 

- You can no longer use fireballs to put people to sleep in order to soulstone them. However, fireballs (and all other explosions) will now reliably stun targets hit, instead of randomly leaving them up sometimes.

- Antags with antistun mechanics can use them against explosive stun, instead of being knocked out and unable to do anything about it.

- Syndicate hardsuit and wizard robes provide better resistance to explosions, but grenade/bombs are still significantly dangerous.

- Bomb suits are now worth consideration for security when facing explosive users, especially when combined with slime speed potion. They provide significant damage resistance, will negate or massively reduce stun duration, and prevent gibbing for max severety explosions.

## Why It's Good For The Game

Explosions are rarely used as a tool due to server rules. 
However, when they ARE used, they are currently frustrating for both parties : Explosions are unreliable as a crowd control tool since the chance to knock out isnt guaranteed, and explosions have little counters because bomb armor currently barely does anything, and you can't use any antistun/escape ability when knocked out.
This PR should hopefully make explosions more fun and less frustrating for both exploder and explodee.

## Changelog
:cl:
tweak: Explosions, including wizard fireballs, no longer knock out, but stun instead.
tweak: Explosion stun is now guaranteed, but the duration can be lowered by wearing armor with bomb resistance.
tweak: Explosion stun duration is further reduced if you are on the edge of the blast zone.
tweak: Wearing armor with bomb resistance will now lower the chance of losing limbs in explosions. Bomb suit negate limb loss entirely.
tweak: Limb loss chance has been reduced if you are on the edge of the blast zone.
tweak: Wearing armor with bomb resistance now reliably lowers the damage caused by explosions, up to a half taken for a complete bomb suit.
/:cl:

### CODE REVIEW

**Review by Kyep — CHANGES_REQUESTED** [review]
Most of this is okay, but I don't think explosions should do less damage to humans.
Other then that, it seems reasonable.

**Comment by datlo** [comment]
My reasoning for lowering the high severity damage from 120 damage to 100 damage is that it makes bomb armor more efficient at preventing a single bomb from putting you into soft crit. I can revert it back to 120 if you prefer. Devastation and low severity damage values are still the same.

**Comment by Kyep** [comment]
@datlo yeah, if you undid the nerf to explosion damage, I'd probably be fine with this and vote for it.

**Comment by datlo** [comment]
Reverted the damage nerf on high severity as requested

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -200,91 +200,60 @@
...
+	var/bomb_armor = getarmor(null, "bomb")
+	var/list/valid_limbs = list("l_arm", "l_leg", "r_arm", "r_leg")
+	var/limbs_amount = 1
+	var/limb_loss_chance = 50
 
 	switch(severity)
-		if(1)
-			b_loss += 500
+		if(EXPLODE_DEVASTATE)
 			if(!prob(getarmor(null, "bomb")))
```
Comment (dearmochi) [inline code/modules/mob/living/carbon/human/human.dm:215]:
```suggestion
			if(!prob(bomb_armor))
```

**Review by dearmochi — APPROVED** [review]

**Review by Kyep — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
